[yiwern5] iP

yiwern5 opened the pull request:
# Iggly

<img src="https://github.com/nus-cs2103-AY2324S2/ip/assets/111892633/e96fd13a-88aa-4fae-9584-54f06162eb4a" alt="Icon" width="200">

> This is Iggly, your personal assistant chatbot! 🐧

---
## Why Iggly?
- easy to use
- ~~cute~~ **SUPER CUTE**

---
## How to use Iggly?

1. Download Iggly [here](https://github.com/yiwern5/ip/releases/download/v0.1/Iggly.jar).
2. Navigate to your Downloads folder using the terminal.
3. Type `java -jar Iggly.jar` in your terminal.
4. Press enter.
5. Give it a try now!

---
## Features
- [X] Add tasks (todo, event and deadline)
- [X] List out all tasks
- [X] Mark the task as done or not done
- [X] Delete a task
- [X] Find tasks based on keyword
- [ ] Filter task (coming soon)
- [ ] Reminder (coming soon)

---
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Duke {
    public static void main(String[] args) {
        Launcher.main(args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

yiwern5 committed:
Level 0
- created duke package
- created Ui.java (logo, greet, exit)

yiwern5 committed:
Level 1
- created command package
- created Command.java (getCommand)

yiwern5 committed:
Level 2
- created Model-View-Controller Structure
- Model (Task)
- Controller (AddTask, ListTask, Greet, Exit, HandleUserInput)
- View (AddTaskView, ListTaskView, GreetView, ExitView)

yiwern5 committed:
Level 2
- created Model-View-Controller Structure
- Model (Task)
- Controller (AddTask, ListTask, Greet, Exit, HandleUserInput)
- View (AddTaskView, ListTaskView, GreetView, ExitView)

yiwern5 committed:
Level 3
- Controller (MarkTask, UnmarkTask)
- View (MarkTaskView, UnmarkTaskView)

yiwern5 committed:
Level 4
- Model (ToDo, Event, Deadline)

yiwern5 committed:
A-TextUiTesting

yiwern5 committed:
Level 5
- Model (DukeException)

yiwern5 committed:
Level 6
- Controller (DeleteTask)
- View (DeleteTaskView)

yiwern5 committed:
Level 7
- Storage.java

yiwern5 committed:
Merge pull request #1 from yiwern5/branch-Level-7

Level 7

yiwern5 committed:
Level 8
- Date and Time for Deadline.java and Event.java

yiwern5 committed:
Merge pull request #2 from yiwern5/branch-Level-8

Level 8

yiwern5 committed:
A-moreOOP
- made all commands to be a subclass of Command.java
- created Parser.java

yiwern5 committed:
A-Enums
- used enums for command string

yiwern5 committed:
Merge branch 'master' of https://github.com/yiwern5/ip into add-gradle-support

yiwern5 committed:
Added checkstyle and used gradle to run for checkstyle test

yiwern5 committed:
A-Javadoc

yiwern5 committed:
A-Javadoc

yiwern5 committed:
Merge pull request #3 from yiwern5/add-gradle-support

A-Gradle

yiwern5 committed:
A-Packages

yiwern5 committed:
A-JUnit

yiwern5 committed:
Merge pull request #4 from yiwern5/branch-A-JUnit

A-JUnit

yiwern5 committed:
A-JAR

yiwern5 committed:
A-JavaDoc

yiwern5 committed:
A-CodingStandard
- changed the name of the test method

yiwern5 committed:
Level-9
- Command (added FindTaskCommand)
- View (added FindTaskView)

yiwern5 committed:
A-JavaDoc

yiwern5 committed:
Merge pull request #5 from yiwern5/branch-A-CodingStandard

A-CodingStandard

yiwern5 committed:
A-CodingStandard

yiwern5 committed:
Level-9

yiwern5 committed:
Level-9

yiwern5 committed:
Level-9

yiwern5 committed:
Minor change to Task.java

yiwern5 committed:
A-CheckStyle

yiwern5 committed:
Merge pull request #6 from yiwern5/branch-A-CheckStyle

A-CheckStyle

yiwern5 committed:
Level-10

yiwern5 committed:
Merge branch 'branch-Level-10'

yiwern5 committed:
JAR file

yiwern5 committed:
JAR file

mahadhir247 commented on src/main/java/iggly/Duke.java:
```diff
@@ -0,0 +1,103 @@
+
+    /**
+     * Constructor for {@link Duke} to initialize the storage or
+     * get task list from the storage.
```
Should there a bit an empty line here to separate the parameter and description to improve readability?

yiwern5 replied:
Alright! Will add a line between parameter and description, thanks for the suggestion :)

mahadhir247 commented on src/main/java/iggly/command/HelpCommand.java:
I like how you added a help command for better UX 👍

maiyasaliha replied:
Yes! This is a very cool Idea! Good thinking :)

mahadhir247 commented on src/main/java/iggly/duke/Parser.java:
```diff
@@ -0,0 +1,219 @@
+        String[] splitTask = command.split(" ", 2);
+        Task task;
+        Command command = null;
+        switch(splitTask[0]) {
```
Good job on making sure the case clauses are not indented 👍

mahadhir247 commented on src/main/java/iggly/command/MarkTaskCommand.java:
Any reason why you decided to split marking/unmarking task? Is it reasonable to try and combine these two methods?

yiwern5 replied:
Hmm I think they should be separated based on single responsibility principle (SRP), as marking and unmarking are 2 distinct responsibilities.

mahadhir247 reviewed (commented):
LGTM. I like how your javadocs is very detailed and clear. Your code also follows coding standards very well. Just a few nits like separation of description from parameters in your javadocs.

maiyasaliha commented on src/main/java/iggly/command/FindTaskCommand.java:
```diff
@@ -0,0 +1,39 @@
+     * prints a filtered {@link TaskList}.
+     *
+     * @param storage The storage object that manages the data persistence. It is not used in {@link FindTaskCommand}
+     */
```
There are clear JavaDocs for almost all methods so far! Makes it very easy to comprehend. Keep up the effort👍🏽

maiyasaliha commented on src/main/java/iggly/duke/Storage.java:
```diff
@@ -0,0 +1,87 @@
+     *
+     * @return {@code true} if the file exists, {@code false} otherwise.
+     */
+    public boolean isFileExists() {
```
isExistingFile( ) might be a better alterative. You're not required to change! Just according to the code quality, this order might be more "sensible" as the textbook puts it.

maiyasaliha commented on src/main/java/iggly/duke/Storage.java:
```diff
@@ -0,0 +1,87 @@
+     *
+     * @param taskList The list of tasks to be written to the file.
+     */
+    public void update(ArrayList<Task> taskList) {
```
updateList maybe? Again by the textbook: Update what? Could be clearer for comprehension upon first look!

yiwern5 replied:
okie thanks for the heads up!

maiyasaliha commented on src/main/java/iggly/model/Deadline.java:
```diff
@@ -0,0 +1,35 @@
+     *
+     * @return A formatted string representing the deadline task.
+     */
+    @Override
+    public String toString() {
+        return "[D]"
+                + super.toString()
+                + " (by: " + date.format(DateTimeFormatter.ofPattern("dd MMM yyyy h:mma"))
+                + ", " + date.getDayOfWeek() + ")";
+    }
```
I did this same thing in mine! I did get a feedback however that it might be too long. Maybe it would be better to extract the formatters into another method!

yiwern5 replied:
hmm true, i think would be better to extract it out!

maiyasaliha commented on src/main/java/iggly/model/Event.java:
```diff
@@ -0,0 +1,38 @@
+     *
+     * @return A formatted string representing the event task.
+     */
+    @Override
+    public String toString() {
+        return "[E]"
+                + super.toString()
+                + " (on: " + from.format(DateTimeFormatter.ofPattern("dd MMM yyyy h:mma"))
+                + " to " + to.format(DateTimeFormatter.ofPattern("h:mma")) + ")";
+    }
```
Here too! Just follow it up with all other occurrences.

maiyasaliha reviewed (commented):
Great Job! Your code actually looks beautiful and makes me **want** to read it, it's very easy on the eyes and to comprehend too! You've done a really good job with Iggly! Just a few extremely minor things you might want to change. But other than that LGTM! Keep up the great work :) Iggly is just the cutest🩷

yiwern5 committed:
PR Feedback

yiwern5 committed:
A-Assertions
Added assertions in Duke.java to ensure the file path of storage and error message is not null.

yiwern5 committed:
A-CI
Added gradle.yml to allow github actions to run the workflow

yiwern5 committed:
BCD-Extension
Added C-BetterSearch by modifying the find method to make it case insensitive

yiwern5 committed:
BCD-Extension

Added B-DoWithinPeriodTasks

yiwern5 committed:
CodeQuality

Made changes based on feedback

yiwern5 committed:
A-UserGuide

Added user guide

yiwern5 committed:
A-UserGuide docs

Update user guide README.md
